js/models/conversations.js

```diff
@@ -66,8 +66,19 @@
             this.messageCollection.get(message.id).fetch();
         }
 
-        return this.getUnread().then(function(unreadMessages) {
-            this.save({unreadCount: unreadMessages.length});
+        // We mark as read everything older than this message - to clean up old stuff
+        //   still marked unread in the database. If the user generally doesn't read in
+        //   the desktop app, so the desktop app only gets read receipts, we can very
+        //   easily end up with messages never marked as read (our previous early read
+        //   receipt handling, read receipts never sent because app was offline)
+
+        // We queue it because we often get a whole lot of read receipts at once, and
+        //   their markRead calls could very easily overlap given the async pull from DB.
+
+        // Lastly, we don't send read receipts for any message marked read due to a read
+        //   receipt. That's a notification explosion we don't need.
+        this.queueJob(function() {
+            return this.markRead(message.get('received_at'), {sendReadReceipts: false});
```

Comment:
You'll need to return a promise from markRead if you want other queued jobs to block on it.